fip-0026.md

```diff
@@ -126,199 +127,267 @@ Holds the public key of the account that will be used to hold all the domains wh
+|fee_collected|Int|Amount of SUFs collected as fee|
 
 ##### Example
 ```
 {
-    "actor": "2odzomo2v4pe",
-    "fio_domain": "alice"
+  "status": "OK",  
+  "buy_price": 100000000000,
+  "fee_collected": 2000000000
 }
 ```
 
-#### Processing
-
-- Request is validated per exception handling
-- Calculate expiration date by taking current expiration date and adding on `duration` time. 
-- Write new expiration date to table
-
-#### Exception handling
-|Error condition|Trigger|Type|fields:name|fields:value|Error message|
-|---|---|---|---|---|---|
-|Invalid FIO domain|Supplied `fio_domain` is not listed for sale.|400| | |"FIO domain not found"|
-|Domain will expire within `duration`|Domain will expire before new listing expiration date|400| | |"Domain will expire before next expiration of listing"|
-|Outside `warn_time` window|The listing does not expire within `warn_time` days.|400| | |"Too early to renew listing"|
-|Actor not signer|The signer and the `actor` do not match |403| | |Type: invalid_signature|
-
-#### Response
-TBD
-
-##### Example
-TBD
-
 ### Set Holder Account
-
+#### Contract: fio.escrow
+#### New fee: set_holder_account, not bundle eligible
+#### New Endpoint: set_holder_account
+#### Ram Increase: To be determined during implementation
 #### New action: *sethldacct*
-This can only be called from the **EscrowContract** or a **SystemAccount**. It will be set once and never need to be changed.
+This can only be called from the **EscrowContract** or a **SystemAccount**. It will be set once and likely will never need to be changed.
 
 #### Request
 |Parameter|Required|Format|Definition|
 |---|---|---|---|
 |public_key|Yes|string|Valid FIO public key of an actor registered on FIO|
+|max_fee|Yes|Positive Int|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by [/get_fee](https://developers.fioprotocol.io/api/api-spec/reference/get-fee/get-fee) for correct value.|
+|tpid|Yes|FIO Address|FIO Address of the entity which generates this transaction. TPID rewards will be paid to this address. Set to empty if not known.|
+|actor|Yes|FIO account|Name of account to pay fee|
 
 ##### Example
 ```
 {
-    "public_key": "FIO7VND52g9gwoahPAbA7gLosbpNY4uKsqy6qFUJysw8gSG15QXWC"
+    "public_key": "FIO7VND52g9gwoahPAbA7gLosbpNY4uKsqy6qFUJysw8gSG15QXWC",
+    "max_fee": 1000000000,
+    "tpid": "rewards@wallet"
 }
 ```
 #### Processing
 - Request is validated per exception handling
 - Write key to 0th index in the table since there will only ever be one key.
+- Verify that the fee for this does not exceed the max fee specified.
+- Charge appropriate fee (this will be a bundled fee transaction, fee will be same as reject)
+- Verify tx does not exceed max transaction size.
 
 #### Exception handling
 |Error condition|Trigger|Type|fields:name|fields:value|Error message|
 |---|---|---|---|---|---|
+|Invalid public key|Invalid FIO Public Key|400|"public_key"|Value sent in, e.g."FIOinValidKey1234232"|"Invalid Public Key"|
```

Comment:
Need to define somewhere what happens if the public key is valid, but not an account.